python/arcticdb/version_store/library.py

```diff
@@ -961,18 +969,27 @@ def sort_and_finalize_staged_data(
         prune_previous_versions : bool, default=False
             Removes previous (non-snapshotted) versions from the database.
 
+        metadata : Any, default=None
+            Optional metadata to persist along with the symbol.
+
+        Returns
+        -------
+        VersionedItem
+            Structure containing metadata and version number of the written symbol in the store.
+            The data member will be None.
+
         See Also
         --------
         write
             Documentation on the ``staged`` parameter explains the concept of staged data in more detail.
         """
-
-        self._nvs.version_store.sort_merge(
+        vit = self._nvs.version_store.sort_merge(
             symbol,
-            None,
+            normalize_metadata(metadata),
```

Comment:
Elsewhere in `_store.py` `normalize_metadata` is only called if `metadata is not None`